pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/InMemorySendingMailbox.java

```diff
@@ -70,11 +74,12 @@ public void send(TransferableBlock block)
 
   @Override
   public void complete() {
+    _isTerminated = true;
   }
 
   @Override
   public void cancel(Throwable t) {
-    if (_isTerminated) {
+    if (_isEarlyTerminated || _isTerminated) {
```

Comment:
We should not change this. Cancel should be applied even if early terminate is called